### PR TITLE
fix feature derivation predict and add tests

### LINKED ISSUE #1156: xgboost predict failed with feature derivation
Predict SQL:

``` sql
SELECT * FROM boston.test
TO PREDICT boston.predict.medv
USING sqlflow_models.my_xgb_regression_model;
```

Train SQL:

```sql
SELECT * FROM boston.train
TO TRAIN xgboost.gbtree
WITH objective="reg:squarederror"
           train.num_boost_round=30
LABEL medv
INTO sqlflow_models.my_xgb_regression_model;
```

logs: 

``` text
==========Program======

import json
import xgboost as xgb
import numpy as np
from sqlflow_submitter.db import connect_with_data_source, db_generator, buffered_db_writer
from sqlflow_submitter.xgboost.predict import pred

feature_field_meta = json.loads('''null''')
label_field_meta = json.loads('''{"name":"medv","dtype":0,"delimiter":"","shape":null,"is_sparse":false,"vocabulary":null,"MaxID":0}''')

pred(datasource='''mysql://root:root@tcp(127.0.0.1:3306)/?maxAllowedPacket=0''',
     select='''SELECT *
FROM boston.test''',
     feature_field_meta=feature_field_meta,
     label_field_meta=label_field_meta,
     result_table='''boston.predict''',
     hdfs_namenode_addr='''''',
     hive_location='''''',
     hdfs_user='''''',
     hdfs_pass='''''')

=======Program Output===========
Traceback (most recent call last):
  File "<stdin>", line 20, in <module>
  File "/go/src/sqlflow.org/sqlflow/python/sqlflow_submitter/xgboost/predict.py", line 30, in pred
    feature_column_names = [k["name"] for k in feature_field_meta]
TypeError: 'NoneType' object is not iterable
```

### PR DESCRIPTION
**Pull request by typhoonzero**
Fix https://github.com/sql-machine-learning/sqlflow/issues/1156

### CODE REVIEW

File: pkg/sql/executor.go [hunk 1]
```diff
@@ -643,6 +648,65 @@ func createPredictionTable(predParsed *extendedSelect, db *DB, session *pb.Sessi
 	return nil
 }
 
+// Create prediction table using the `PredictIR`.
+// TODO(typhoonzero): remove legacy `createPredictionTable` if we moved all to IR.
```
Comment (terrytangyuan) [inline pkg/sql/executor.go]: if we moved -> once we move

File: cmd/sqlflowserver/main_test.go [hunk 2]
```diff
@@ -290,6 +290,30 @@ func TestEnd2EndMySQL(t *testing.T) {
 	// Cases using feature derivation
 	t.Run("CaseTrainTextClassificationIR", CaseTrainTextClassificationIR)
 	t.Run("CaseTrainTextClassificationFeatureDerivation", CaseTrainTextClassificationFeatureDerivation)
+	t.Run("CaseXgboostFeatureDerivation", CaseXgboostFeatureDerivation)
+	t.Run("CaseTrainFeatureDerevation", CaseTrainFeatureDerevation)
+}
+
+func CaseXgboostFeatureDerivation(t *testing.T) {
```
Comment (Yancey0623) [inline cmd/sqlflowserver/main_test.go:297]: Xgboost => XGBoost

**Review by Yancey0623 — APPROVED** [review]
LGTM

File: cmd/sqlflowserver/main_test.go (hunk 2 above)
Comment (typhoonzero) [reply cmd/sqlflowserver/main_test.go:297]: Will refine in next PR.

**Pull request merged**